### PR TITLE
Added Navitron JS and base plugin CSS

### PR DESCRIPTION
**Pull request by mlworks**
Set up JS functionality and base plugin CSS

Status: **Opened for visibility**

Reviewers: @donnielrt @jeffkamo  
## Changes
- Added JS for setting up and interacting with Navitron
- Added base CSS for plugin
- Set up example page with styles
## How to test-drive this PR
- Copy repo to desktop
- npm install, bower install, grunt serve
- Go to http://localhost:3000

### CODE REVIEW

File: dist/navitron.js [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+                {
+                    display: 'none',
+                    easing: this.options.easing,
+                    duration: this.options.duration,
+                    mobileHA: this.options.mobileHA,
+                    begin: function() {
+                        // Setting z-index to make sure pane shifting out will always be above
+                        // the pane that's being shifted back in
+                        $currentMenu.css({
+                            zIndex: 2
```
Comment (kpeatt) [inline dist/navitron.js]: Are you concerned about this hard set z-index? Should this be either relative to an existing element or has this worked fine for you?

**Comment by tedtate** [comment]
@donnielrt @jeffkamo would you guys mind taking a look at this?

File: src/style/navitron.scss [hunk 2]
```diff
@@ -1,4 +1,85 @@
...
+//
+//    [Note] Must keep this as 'display: none' otherwise you can
+//    crash the browser translating too many divs at once. This
+//    also helps with Pinny's opening/closing animation.
+//
+// 2. Used for unique sliding effects. For example: Flipping
+//    the next pane over like a page.
+
+.navitron__nested {
+    position: absolute;
```
Comment (jeffkamo) [inline src/style/navitron.scss]: What is this absolutely positioned to? (I'm assuming `.navitron`, but would be nice to be explicit)

File: src/style/navitron.scss [hunk 3]
```diff
@@ -1,4 +1,85 @@
...
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    z-index: 10;
```
Comment (jeffkamo) [inline src/style/navitron.scss]: Why does this need `z-index`?

File: src/style/navitron.scss [hunk 4]
```diff
@@ -1,4 +1,85 @@
...
+// that are offsetted outside of Navitron wrapper.
+//
+//
+// Notes:
+//
+// 1. Hide nested sliding panes by default.
+//
+//    [Note] Must keep this as 'display: none' otherwise you can
+//    crash the browser translating too many divs at once. This
+//    also helps with Pinny's opening/closing animation.
```
Comment (jeffkamo) [inline src/style/navitron.scss]: Should we be mentioning other plugins in the comments?

File: src/style/navitron.scss [hunk 5]
```diff
@@ -1,4 +1,85 @@
...
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    z-index: 10;
+
+    transform: translate3d(100%, 0, 0);
+
+    perspective: 1000px; // 2
```
Comment (jeffkamo) [inline src/style/navitron.scss]: Is there a reason `1000px` was specifically chosen for the value?

File: src/style/navitron.scss [hunk 6]
```diff
@@ -1,4 +1,85 @@
...
+//
+// This is the sliding pane that gets animated in and out of view.
+//
+//
+// Notes:
+//
+// 1. Avoid content bleed through.
+
+.navitron__pane {
+    position: absolute;
```
Comment (jeffkamo) [inline src/style/navitron.scss]: Same comment as above, it might be nice to explicitly mention what this is absolutely positioned relative to.

Comment (jeffkamo) [inline src/style/navitron.scss]: In fact, if this positioning is being applied to both `.navitron__nested` and `.navitron__pane`, would it be worth declaring them once in separate block?

``` scss
.navitron__nested,
.navitron__pane {
    position: absolute;
    // ...
}
```

**Comment by donnielrt** [comment]
Hey, we’re looking to prune older unattended PRs. If this PR is still relevant and you would like to see it merged in, please reopen the PR and we’ll add it to our backlog! Thanks!

**Pull request closed without merging**